azurerm/resource_arm_mariadb_firewall_rule.go

```diff
@@ -0,0 +1,154 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/mariadb/mgmt/2018-06-01/mariadb"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMariaDBFirewallRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMariaDBFirewallRuleCreateUpdate,
+		Read:   resourceArmMariaDBFirewallRuleRead,
+		Update: resourceArmMariaDBFirewallRuleCreateUpdate,
+		Delete: resourceArmMariaDBFirewallRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"server_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
And some additional validation here too

Reply:
there is existing validation on the server resource that you could move into a helper function and share

Reply:
I made helper functions for validating server name and firewall rule names. Should I open a separate PR where I'd make the mariadb_server resource use the same validation function aswell?